### PR TITLE
Implemented method `getStatistic(String fromFileName, String toFileName)

### PR DESCRIPTION
**Pull request by Jacalius**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String RESULT_SUPPLY = "supply";
+    private static final String RESULT_BUY = "buy";
+    private static final String RESULT_RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:16]: [Divide program logic](https://github.com/mate-academy/jv-work-with-file/blob/master/checklist.md#divide-program-logic)
Let's create separate methods: `readFile(fromFileName); getResult(output); writeToFile(result, toFileName);`

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String RESULT_SUPPLY = "supply";
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:12]: Pointless duplication, remove 
```
private static final String RESULT_SUPPLY = "supply";
private static final String RESULT_BUY = "buy";
```

Add constant for "comma" and all magic numbers, that you use

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,47 @@
...
+                        countSupply += amount;
+                    }
+                    if (operation.equals(OPERATION_BUY)) {
+                        countBuy += amount;
+                    }
+                }
+            }
+            int result = countSupply - countBuy;
 
+            bufferedWriter.write(RESULT_SUPPLY + "," + countSupply);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:38]: For creating report is better to use StringBuilder.append()

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,47 @@
...
+            bufferedWriter.newLine();
+            bufferedWriter.write(RESULT_BUY + "," + countBuy);
+            bufferedWriter.newLine();
+            bufferedWriter.write(RESULT_RESULT + "," + result);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file:", e);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:44]: Let's improve your error message see the [link](https://github.com/mate-academy/jv-work-with-file/blob/master/checklist.md#dont-ignore-exceptions)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:10]: create also such constants: but, supply, result

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,66 @@
...
+        String result = getResult(report);
+        writeToFile(result, toFileName);
+    }
+
+    private String readFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                stringBuilder.append(line).append("\n");
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:23]:
```suggestion
                stringBuilder.append(line).append(System.lineSeparator());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,7 +1,69 @@
...
+    private String getResult(String report) {
+        int countSupply = 0;
+        int countBuy = 0;
+
+        String[] lines = report.split(System.lineSeparator());
+        for (String line : lines) {
+            String[] parts = line.split(COMMA);
+            if (parts.length == 2) {
+                String operation = parts[0].trim();
+                int amount = Integer.parseInt(parts[1].trim());
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:43]: extract to OPERATION_INDEX, AMOUNT_INDEX constants

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request closed without merging**